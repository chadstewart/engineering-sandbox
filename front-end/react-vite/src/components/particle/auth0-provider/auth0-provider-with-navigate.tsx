import { Auth0Provider } from "@auth0/auth0-react";

interface Auth0ProviderWithNavigateProps {
  children: React.ReactNode;
}

export const Auth0ProviderWithNavigate = ({ children }: Auth0ProviderWithNavigateProps) => {
  const domain = import.meta.env.VITE_AUTH0_URL;
  const clientId = import.meta.env.VITE_AUTH0_CLIENT_ID;
  const redirectUri = import.meta.env.VITE_AUTH0_CALLBACK_URL;
  const audience = import.meta.env.VITE_AUTH0_AUDIENCE;

  if (!(domain && clientId && redirectUri)) {
    return null;
  }

  return (
    <Auth0Provider
      domain={domain}
      clientId={clientId}
      authorizationParams={{
        redirect_uri: redirectUri,
        audience: audience
      }}
    >
      {children}
    </Auth0Provider>
  );
};
