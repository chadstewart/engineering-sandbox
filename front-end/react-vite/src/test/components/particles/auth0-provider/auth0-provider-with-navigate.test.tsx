/**
 * @vitest-environment happy-dom
 */

import { Auth0ProviderWithNavigate } from "@/components/particle/auth0-provider/auth0-provider-with-navigate";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Particle Component: Auth0ProviderWithNavigate", () => {
  it("Should match the snapshot", () => {
    const {
      container: { firstChild: variableToSnapshot }
    } = render(<Auth0ProviderWithNavigate children={<div></div>} />);

    expect(variableToSnapshot).toMatchSnapshot();
  });
});
