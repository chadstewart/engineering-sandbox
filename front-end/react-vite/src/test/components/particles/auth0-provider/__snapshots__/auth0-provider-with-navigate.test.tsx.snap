// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Particle Component: Auth0ProviderWithNavigate > Should match the snapshot 1`] = `<div />`;
